instrumentation/http-url-connection/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/httpurlconnection/GetOutputStreamContext.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.httpurlconnection;
+
+import static io.opentelemetry.context.ContextKey.named;
+
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.ContextKey;
+import io.opentelemetry.context.ImplicitContextKeyed;
+import java.net.HttpURLConnection;
+
+public class GetOutputStreamContext implements ImplicitContextKeyed {
+  private static final ContextKey<GetOutputStreamContext> KEY =
+      named("opentelemetry-http-url-connection-get-output-stream");
+
+  private boolean outputStreamMethodOfSunConnectionCalled;
+
+  private GetOutputStreamContext() {}
+
+  @Override
+  public Context storeInContext(Context context) {
+    return context.with(KEY, this);
+  }
+
+  public static Context init(Context context) {
+    if (context.get(KEY) != null) {
+      return context;
+    }
+    return context.with(new GetOutputStreamContext());
+  }
+
+  public static GetOutputStreamContext retrieveFrom(Context context) {
+    return context.get(KEY);
+  }
+
+  public void set(
```

Comment:
In other places that use the same pattern we typically have static methods for setting and getting values.

Reply:
Code changed